### ISSUE TITLE
[Reimplementation] loss become infinite or NaN when training SSD512

### ISSUE THREAD

**Issue by cmjkqyl** [open-mmlab/mmdetection#9463]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### 💬 Describe the reimplementation questions

“loss become infinite or NaN” is reported when training SSD512

The dataset is all correct and other models in CONFIGS can work well. 

I have tried to reduce the LR or increase the warmup_iters, but it does not work.

### Environment

CUDA==11.1 
torch==1.8.0
mmcv-full==1.3.8
mmdet==2.18.1

**Comment by RangiLyu** [comment]
What dataset are you using? And what is the total batch size(num_gpus x samples_per_gpu) you used?

**Comment by RangiLyu** [comment]
SSD with VGG backbone is hard to train because it does not apply modern tricks to prevent gradient explosion and vanishing. 
This algorithm requires careful hyperparameter tuning. If you don't have rich experience in tuning hyperparameters, I suggest you change to a relatively new algorithm in recent years such as RetinaNet.

**Comment by JingAo-Shen** [comment]
> SSD with VGG backbone is hard to train because it does not apply modern tricks to prevent gradient explosion and vanishing. This algorithm requires careful hyperparameter tuning. If you don't have rich experience in tuning hyperparameters, I suggest you change to a relatively new algorithm in recent years such as RetinaNet.

thank you, very helpful.
